app/models/spree/payment.rb

```diff
@@ -145,7 +145,7 @@ def ensure_correct_adjustment
         adjustment.originator = payment_method
         adjustment.label = adjustment_label
         adjustment.save
-      else
+      elsif payment_method.present?
```

Comment:
I think we should drop this change for now as we're not covering it with tests here and it modifies a key bit of logic in the `Payment` model.

Reply:
(I cc @andrewpbrett because we are two on this PR)